### PR TITLE
feat(scheduler-targets-alpha): `InspectorStartAssessmentRun` Target

### LINKED ISSUE #27453: (aws-scheduler-targets-alpha): Add InspectorStartAssessmentRun Target
### Describe the feature

Work to support L2 constructs for AWS Scheduler is in progress (https://github.com/aws/aws-cdk/issues/23394). See the approved [RFC](https://github.com/aws/aws-cdk-rfcs/pull/486/files). RFC planned to add 12 templates targets, but only Lambda Invoke is currently implemented (https://github.com/aws/aws-cdk/pull/26575). 

This issue tracks implementation of  `InspectorStartAssessmentRun` target to start an AWS Inspector assessment run.

### Use Case

Customers would like to use templated target `InspectorStartAssessmentRun` to be able to start a record to an AWS Inspector assessment run on schedule. L2 target construct should grant required permissions to the `AWS Scheduler` to start an AWS Inspector Assessment run.

### Proposed Solution

The proposed solution needs to be adopted to the recent examples of `LambdaInvoke` (https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/aws-scheduler-targets-alpha/lib/lambda-invoke.ts). 

Solution should also include unit and integration tests. 

```ts
  export class InspectorStartAssessmentRun extends ScheduleTargetBase implements IScheduleTarget {
    constructor(
      private readonly target: inspector.CfnAssessmentTarget,
      private readonly props: ScheduleTargetBaseProps,
    ) {
      super(props, target.attrArn);
    }

    protected addTargetActionToRole(schedule: ISchedule, role: IRole): void {
     // TODO: Check if target and schedule are in the same account and region
     // TODO: Check if target and role are in the same account

     // TODO: Check if L2 grant methods can be used
      role.addToPrincipalPolicy(new iam.PolicyStatement({
        actions: ['inpsector:StartAssessmentRun'],
        resources: ["*"],
      }));      
    }
  }
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.99.1

### Environment details (OS name and version, etc.)

MacOS

### PR DESCRIPTION
**Pull request by go-to-k**
This PR adds InspectorStartAssessmentRun Target for EventBridge Scheduler.

In [the issue](https://github.com/aws/aws-cdk/issues/27453), the `inspector.CfnAssessmentTarget` is used in the `InspectorStartAssessmentRun`. But it should be a `CfnAssessmentTemplate` so I fixed.

```ts
  export class InspectorStartAssessmentRun extends ScheduleTargetBase implements IScheduleTarget {
    constructor(
      private readonly target: inspector.CfnAssessmentTarget, // <- here
      private readonly props: ScheduleTargetBaseProps,
    ) {
```

Closes #27453.

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by go-to-k** [comment]
@lpizzinidev 

Thanks! I changed them.

File: packages/@aws-cdk/aws-scheduler-targets-alpha/README.md [hunk 1]
```diff
@@ -121,3 +122,21 @@ new Schedule(this, 'Schedule', {
...
+called every hour by Event Bridge Scheduler.
+
+```ts
+import * as inspector from 'aws-cdk-lib/aws-inspector';
+
+declare const assessmentTemplate: inspector.CfnAssessmentTemplate;
+
+new Schedule(this, 'Schedule', {
+  schedule: ScheduleExpression.rate(Duration.minutes(60)),
+  target: new targets.InspectorStartAssessmentRun(assessmentTemplate, {}),
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-scheduler-targets-alpha/README.md:140]:
```suggestion
  target: new targets.InspectorStartAssessmentRun(assessmentTemplate),
```
Properties are now optional, let's keep the documentation more concise.

File: packages/@aws-cdk/aws-scheduler-targets-alpha/test/integ.inspector-start-assessment-run.ts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const assessmentTarget = new CfnAssessmentTarget(stack, 'MyAssessmentTarget');
+const assessmentTemplate = new CfnAssessmentTemplate(stack, 'MyAssessmentTemplate', {
+  assessmentTargetArn: assessmentTarget.attrArn,
+  durationInSeconds: 3600,
+  rulesPackageArns: ['arn:aws:inspector:us-east-1:316112463485:rulespackage/0-gEjTy7T7'],
+});
+
+new scheduler.Schedule(stack, 'Schedule', {
+  schedule: scheduler.ScheduleExpression.rate(cdk.Duration.minutes(10)),
+  target: new InspectorStartAssessmentRun(assessmentTemplate, {}),
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-scheduler-targets-alpha/test/integ.inspector-start-assessment-run.ts:24]:
```suggestion
  target: new InspectorStartAssessmentRun(assessmentTemplate),
```

File: packages/@aws-cdk/aws-scheduler-targets-alpha/test/inspector-start-assessment-run.test.ts [hunk 3]
```diff
@@ -0,0 +1,509 @@
...
+    const assessmentTarget = new CfnAssessmentTarget(stack, 'MyAssessmentTarget');
+    template = new CfnAssessmentTemplate(stack, 'MyTemplate', {
+      assessmentTargetArn: assessmentTarget.attrArn,
+      durationInSeconds: 3600,
+      rulesPackageArns: ['arn:aws:inspector:us-east-1:316112463485:rulespackage/0-gEjTy7T7'],
+    });
+  });
+
+  test('creates IAM role and IAM policy for inspector assessment template in the same account', () => {
+    const inspectorTarget = new InspectorStartAssessmentRun(template, {});
```
Comment (lpizzinidev) [inline packages/@aws-cdk/aws-scheduler-targets-alpha/test/inspector-start-assessment-run.test.ts:27]:
```suggestion
    const inspectorTarget = new InspectorStartAssessmentRun(template);
```
Can you please remove the optional parameters here as well? where possible.

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks!
Just some minor adjustment after setting a default value for the `props` parameter and it will be good to go for me.

**Comment by go-to-k** [comment]
@lpizzinidev 

Thanks! I missed the modification. I changed.

**Review by lpizzinidev — APPROVED** [review]
Thanks 👍

**Pull request merged**